cli/crud/import/import.go

```diff
@@ -0,0 +1,1214 @@
+	for uint32(csvReader.NumLine) > ctx._masterPosition {
+		// value of current position in raw reader should catch up with the reading position of the main parser
+		// if a single entry in a CSV file could not make up several lines, then this mechanism would not be needed
+		// example: 123,"da\nta",321 is one csv record, but unparsed form is two lines, parsed form is slice ["123", "da\nta", 321]
+		ctx._masterPosition++
+		ctx._scanner.Scan()
+		ctx._currentRecord += "\n" + ctx._scanner.Text()
+	}
+
+	var currentPosition uint32
+	if ctx._masterPosition != ctx._slavePosition {
+		currentPosition = ctx._slavePosition
+	} else {
+		currentPosition = ctx._masterPosition
+	}
+	currentRec := ctx._currentRecord
+	ctx._currentRecord = ""
+
+	return currentPosition, currentRec
+}
+
+var importSummary t_importSummary = t_importSummary{}
+
+// RunImport implements data import via crud.
+func RunImport(cmdCtx *cmdcontext.CmdCtx, conn *connector.Conn, csvReader *ttcsv.Reader, rawReaderFd *os.File, crudImportFlags *Opts, spaceName string) error {
+	sigIntInput := make(chan os.Signal, 1)
+	signal.Notify(sigIntInput, os.Interrupt, syscall.SIGTERM)
+	sigIntInterceptor(sigIntInput)
+
+	// init progress file context
+	var progressCtx *t_progressCtx = &t_progressCtx{
+		StartTimestamp:    time.Now().String(),
+		LastDumpTimestamp: "",
+		EndOfFileReached:  false,
+		LastPosition:      1,
+		RetryPosition:     []uint32{},
+	}
+
+	// init dump subsystem context
+	dumpSubsystemFd, initDumpFderr := initDumpSubsystemFd(
+		crudImportFlags.LogFileName+".log",
+		crudImportFlags.ErrorFileName+".csv",
+		crudImportFlags.SuccessFileName+".csv",
+		progressCtx)
+	if initDumpFderr != nil {
+		return initDumpFderr
+	}
+	defer dumpSubsystemFd.logFile.Close()
+	defer dumpSubsystemFd.errorFile.Close()
+	defer dumpSubsystemFd.successFile.Close()
+
+	// init raw parser context
+	unparsedCsvReaderCtx := &t_unparsedCsvReaderCtx{
+		_masterPosition: 0,
+		_currentRecord:  "",
+		_slavePosition:  0,
+		_scanner:        bufio.NewScanner(rawReaderFd),
+	}
+
+	if evalsInitErr := initCrudImportSessionStorageEvals(conn); evalsInitErr != nil {
+		return evalsInitErr
+	}
+
+	if initTargetSpaceErr := setTargetSpace(conn, spaceName); initTargetSpaceErr != nil {
+		return initTargetSpaceErr
+	}
+
+	if initCrudOperationErr := setCrudOperation(conn, crudImportFlags.Operation); initCrudOperationErr != nil {
+		return initCrudOperationErr
+	}
+
+	if initNullVallErr := setNullInterpretation(conn, crudImportFlags.NullVal); initNullVallErr != nil {
+		return initNullVallErr
+	}
+
+	fmt.Printf("In case of error:\t[%s]\n", crudImportFlags.OnError)
+	fmt.Printf("PID of this process:\t[%d]\n", os.Getpid())
+	fmt.Printf("\n\033[0;33mWARNING: Process is not sensitive to SIGINT (ctrl+c), use kill -9 %d\033[0m\n", os.Getpid())
+
+	var batchSequenceCtx *t_batchSequenceCtx
+	var batch *t_batch
+	if crudImportFlags.Header {
+		/*
+			initialization in the case of a header use
```

Comment:
Use `//' for comments everywhere.
Here and below.

Reply:
Done